src/development/component.md

```diff
@@ -0,0 +1,280 @@
+# Implementing a Component
+
+Each Tock [board](https://github.com/tock/tock/tree/master/boards) defines the
+peripherals, capsules, kernel settings, and syscall drivers to customize Tock
+for that board. Often, instantiating different resources (particularly capsules
+and drivers) requires subtle setup steps that are easy to get wrong. The setup
+steps are often shared from board-to-board. Together, this makes configuring a
+board redundant and easy to make a mistake.
+
+Components are the Tock mechanism to help address this. Each component includes
+the static memory allocations and setup steps required to implement a particular
+piece of kernel functionality (i.e. a capsule). You can read more technical
+documentation [here](https://docs.tockos.org/kernel/component/).
+
+In this guide we will create a component for a hypothetical system call driver
+called `Notifier`. Our system call driver is going to use an alarm as a resource
+and requires just one other parameter: a delay value in milliseconds. The steps
+should be the same for any capsule you want to create a component for.
+
+## Setup
+
+This guide assumes you already have the capsule created, and ideally that you
+have set it up with a board to test. Making a component then just makes it
+easier to include on a new board and share among boards.
+
+## Overview
+
+The high-level steps required are:
+
+1. Define the static memory required for all objects used.
+2. Create a struct that holds all of the resources and configuration necessary
+   for the capsules.
+3. Implement `finalize()` to initialize memory and perform setup.
+
+## Step-by-Step Guide
+
+The steps from the overview are elaborated on here.
+
+1. **Define the static memory required for all objects used.**
+
+   All objects in the kernel are statically allocated, so we need to statically
+   allocate memory for the objects to live in. Now, the size of those objects
+   depends on the specific board the kernel is being compiled for. Therefore, we
+   must use a macro so that all of the types are known in advance to enable the
+   kernel to compile successfully.
```

Comment:
```suggestion
   allocate memory for the objects to live in. Due to [constraints](https://github.com/tock/tock/blob/b41ecd33a361d90820df1e290086d1b22e192e54/kernel/src/utilities/static_init.rs#L56-L80) on the macros
   Tock provides for statically allocating memory, we must contain all calls to
   allocate this memory within another macro.
```

I prefer this shorter explanation, since there are technically several reasons we need to use a macro, and cannot use generics (which would otherwise seem a suitable alternative to a macro based on the description currently provided)